src/libs/actions/IOU.ts

```diff
@@ -4332,9 +4332,14 @@ function getUpdateMoneyRequestParams(
         (hasModifiedTag || hasModifiedCategory || hasModifiedComment || hasModifiedDistanceRate || hasModifiedAmount || hasModifiedCreated)
     ) {
         const currentTransactionViolations = allTransactionViolations[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`] ?? [];
+        const hasModifiedDate = 'date' in transactionChanges;
```

Comment:
Please add a comment here, something like:
```
// If the amount or date have been modified, we remove the duplicate violations since they would be out of date as the transaction has changed
```

But also, why isn't currency here too?